packages/frontend/src/state/positions/hooks.ts

```diff
@@ -112,15 +113,115 @@ export const useSwaps = () => {
+            //sold Squeeth amount
+            acc.soldSqueeth = acc.soldSqueeth.plus(squeethAmt.abs())
+            //usd value from sell to close long position or open short
+            acc.totalUSDFromSell = acc.totalUSDFromSell.plus(usdAmt.abs())
+          } else if (squeethAmt.isNegative()) {
+            //bought Squeeth amount
+            acc.boughtSqueeth = acc.boughtSqueeth.plus(squeethAmt.abs())
+            //usd value from buy to close short position or open long
+            acc.totalUSDFromBuy = acc.totalUSDFromBuy.plus(usdAmt.abs())
+          }
+          if (acc.squeethAmount.isZero()) {
+            acc.longUsdAmount = BIG_ZERO
+            acc.shortUsdAmount = BIG_ZERO
+            acc.wethAmount = BIG_ZERO
+            acc.boughtSqueeth = BIG_ZERO
+            acc.soldSqueeth = BIG_ZERO
+            acc.totalUSDFromSell = BIG_ZERO
+            acc.totalUSDFromBuy = BIG_ZERO
+          } else {
+            // when the position is partially closed, will accumulate usdamount
+            acc.longUsdAmount = acc.longUsdAmount.plus(usdAmt)
+            acc.shortUsdAmount = acc.shortUsdAmount.plus(usdAmt.negated())
+            acc.wethAmount = acc.wethAmount.plus(wethAmt.negated())
+          }
+          return acc
+        },
+        {
+          squeethAmount: BIG_ZERO,
+          wethAmount: BIG_ZERO,
+          longUsdAmount: BIG_ZERO,
+          shortUsdAmount: BIG_ZERO,
+          boughtSqueeth: BIG_ZERO,
+          soldSqueeth: BIG_ZERO,
+          totalUSDFromBuy: BIG_ZERO,
+          totalUSDFromSell: BIG_ZERO,
+        },
+      ) || {
+        squeethAmount: BIG_ZERO,
+        wethAmount: BIG_ZERO,
+        longUsdAmount: BIG_ZERO,
+        shortUsdAmount: BIG_ZERO,
+        boughtSqueeth: BIG_ZERO,
+        soldSqueeth: BIG_ZERO,
+        totalUSDFromBuy: BIG_ZERO,
+        totalUSDFromSell: BIG_ZERO,
+      },
+    [isWethToken0, data?.swaps, getUsdAmt],
+  )
 
-  if (!context) {
-    throw new Error('useComputeSwaps must be used inside ComputeSwapsProvider')
-  }
+  useAppEffect(() => {
+    if (computedSwaps.squeethAmount.isGreaterThan(0) && oSqueethBal?.isGreaterThan(0)) {
+      setPositionType(PositionType.LONG)
+      // check if user osqth wallet balance is equal to the accumulated amount from tx history
+      // if it's not the same, it's likely that they do smt on crab acution or otc or lp etc so dont show the pnl for them
+      if (!computedSwaps.squeethAmount.isEqualTo(oSqueethBal)) {
+        setIsToHidePnL(true)
+      } else {
+        setIsToHidePnL(false)
+      }
+    } else if (computedSwaps.squeethAmount.isLessThan(0)) {
+      setIsToHidePnL(true)
+      setPositionType(PositionType.SHORT)
+    } else {
+      setIsToHidePnL(false)
+      setPositionType(PositionType.NONE)
+    }
+  }, [computedSwaps.squeethAmount, oSqueethBal, setPositionType, setIsToHidePnL])
 
-  return context
-}
+  useAppEffect(() => {
+    refetch()
+  }, [computedSwaps.squeethAmount, refetch])
+
```

Comment:
@adam5909099  there is this continuous re-rendering that's happening. I think it's coming from these lines 207-209. Cos when I comment those lines, it stops.
https://user-images.githubusercontent.com/40301524/169861910-1179b90d-63b6-4fa4-9d4f-3cdd40a8b779.mp4

Reply:
@adam5909099 Seems to be an issue, can you check this!

Reply:
@EstherSeyi I tried logging in that useEffect callback but it's not called infinitely
![image](https://user-images.githubusercontent.com/59410529/170002430-453f0cf5-efb0-4e02-86d1-5f98c4693573.png)
![image](https://user-images.githubusercontent.com/59410529/170002357-f523eef7-ed43-4e90-a630-87fe8f0a2226.png)
Can you let me know the step for reproducing your recording?

Reply:
It seems that when you have longer txs history, it will cause this loop 
My squeeth amount was not changed here with this address `0x49A8E210967553fd22a0C235d47cFAef321276f5`

https://user-images.githubusercontent.com/26922518/170023318-c0f8c8d8-f686-4239-964b-d629a917e6a3.mov

Reply:
@adam5909099 I wanted to add to my short position, so I was on the short open tab when it was happening.

Reply:
@yichiehliu Please pull the latest of this branch. I can see the console log from providers.tsx, but the providers component doesn't exist anymore

Reply:
@EstherSeyi It was related to polling in useTokenBalance. I resolved this by adding queryRefecth, which handles real refetching.